### ISSUE TITLE
No methods support complex matrices multiplied by a real checkerboard matrix

### ISSUE THREAD

**Issue by TongSericus** [cohensbw/Checkerboard.jl#6]
All mul! functions in the module restrict the element types of B and \Gamma to be the same. I don't know if there could be any performance or algorithmic issues if B is complex while \Gamma is real, but it would be helpful to have such a method in case one wants to use complex HS transforms.

**Comment by cohensbw** [comment]
Interesting point. My thinking originally was that the propagator matrices are defined as either `B = exp(-dtau V) exp(-dtau K)` or `B = exp(-dtau K/2) exp(-dtau V) exp(-dtau K/2)`. In either case `exp(-dtau V)` is diagonal, and therefore must be real (not complex) in order to be Hermitian. Therefore, in any situation where a propagator matrix `B`  and Greens function matrix `G` is complex, so must `exp(-dtau K)` be complex as well, and therefore `Gamma` too. But perhaps there are exceptions I have not considered. My main goal was to make it impossible to multiply a real matrix `G` by `Gamma` when `Gamma` is complex.

**Comment by TongSericus** [comment]
exp(-dtau V) is diagonal but can be complex if one uses a complex HS transform, i.e., the auxiliary field is coupled to the charge sector to preserve the SU(2) symmetry. In such cases, exp(-dtau K) or exp(-dtau K/2) is still real but exp(-dtau V) is complex. Hence, one needs to multiply a real matrix by a complex matrix.

**Comment by cohensbw** [comment]
Good point. I just pushed a new version of the package, v1.2.5, and submitted it to be registered that enables you to multiply a complex matrix my a real checkerboard factorization. I additionally introduced the following assert to prevent users from multiplying a real matrix by a complex checkerboard factorization:
`@assert !(T<:Real && E<:Complex) "Cannot multiply a real valued matrix by a complex checkerboard matrix!"`
It can take a few hours for the new version of the package to get registered, but by tomorrow it should be available for upgrade/install using the package manager in Julia.

**Comment by TongSericus** [comment]
Much appreciated!
